src/main/java/org/prebid/server/bidder/adnuntius/AdnuntiusBidder.java

```diff
@@ -0,0 +1,252 @@
+        final List<HttpRequest<AdnuntiusRequest>> adnuntiusRequests = new ArrayList<>();
+
+        try {
+            for (Imp imp : request.getImp()) {
+                validateImp(imp);
+
+                final ExtImpAdnuntius extImpAdnuntius = parseImpExt(imp);
+
+                final String key = StringUtils.stripToEmpty(extImpAdnuntius.getNetwork());
+                final String auId = extImpAdnuntius.getAuId();
+                networkToAdUnits.computeIfAbsent(key, k -> new ArrayList<>())
+                        .add(AdnuntiusAdUnit.of(auId, auId + TARGET_ID_DELIMITER + imp.getId()));
+            }
+        } catch (PreBidException e) {
+            return Result.withError(BidderError.badInput(e.getMessage()));
+        }
+
+        final AdnuntiusMetaData metaData = createMetaData(request.getUser());
+        final String page = extractPage(request.getSite());
+        final String uri = createUri(request);
+
+        for (List<AdnuntiusAdUnit> adUnits : networkToAdUnits.values()) {
+            final AdnuntiusRequest adnuntiusRequest = AdnuntiusRequest.of(adUnits, metaData, page);
+            adnuntiusRequests.add(createHttpRequest(adnuntiusRequest, uri));
+        }
+
+        return Result.withValues(adnuntiusRequests);
+    }
+
+    private static void validateImp(Imp imp) throws PreBidException {
+        if (imp.getBanner() == null) {
+            throw new PreBidException(String.format("Fail on Imp.Id=%s: Adnuntius supports only Banner", imp.getId()));
+        }
+    }
+
+    private ExtImpAdnuntius parseImpExt(Imp imp) throws PreBidException {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), ADNUNTIUS_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(String.format("Unmarshalling error: %s", e.getMessage()));
+        }
+    }
+
+    private static AdnuntiusMetaData createMetaData(User user) {
+        final String userId = ObjectUtil.getIfNotNull(user, User::getId);
+        return StringUtils.isNotBlank(userId) ? AdnuntiusMetaData.of(userId) : null;
+    }
+
+    private static String extractPage(Site site) {
+        return StringUtils.defaultIfBlank(ObjectUtil.getIfNotNull(site, Site::getPage), DEFAULT_PAGE);
+    }
+
+    private String createUri(BidRequest bidRequest) {
+        try {
+            final URIBuilder uriBuilder = new URIBuilder(endpointUrl)
+                    .addParameter("format", "json")
+                    .addParameter("tzo", getTimeZoneOffset());
+
+            final String gdpr = extractGdpr(bidRequest.getRegs());
+            final String consent = extractConsent(bidRequest.getUser());
+            if (gdpr != null && StringUtils.isNotBlank(consent)) {
+                uriBuilder.addParameter("gdpr", gdpr);
+                uriBuilder.addParameter("consentString", consent);
+            }
+
+            return uriBuilder.build().toString();
```

Comment:
Can we narrow down scope of try-catch only to this line?

Reply:
No :( new URIBuilder() and .build throws checked exception